test/test_um2netcdf.py

```diff
@@ -5,7 +5,209 @@
+                    c.coord["longitude"] = 0.0  # FIXME
+
+                m_iris_load.return_value = cubes
+
+                with mock.patch("iris.fileformats.netcdf.Saver") as m_saver:  # prevent I/O
+                    # mock `sman` var to prevent I/O
+                    m_sman = mock.Mock()
+                    m_saver().__enter__.return_value = m_sman
+
+                    # TODO: fix lat/lon & levels requires c.coord attribute
+                    #       use fixtures to add attrs & remove the patches?
+                    with mock.patch("umpost.um2netcdf.fix_latlon_coord") as m_coord:
+                        with mock.patch("umpost.um2netcdf.fix_level_coord") as m_level:
+                            with mock.patch("umpost.um2netcdf.apply_mask") as m_apply_mask:
+                                with mock.patch("umpost.um2netcdf.cubewrite") as m_cubewrite:
+                                    infile = "/tmp/fake_input_fields_file"
+                                    outfile = "/tmp/fake_input_fields_file.nc"
+
+                                    std_args.verbose = True  # test some warning branches
+                                    um2nc.process(infile, outfile, std_args)
+
+                                    assert m_sman.update_global_attributes.called
+                                    assert m_saver.write.called is False  # write I/O prevented
+                                    assert m_coord.called
+                                    assert m_level.called
+                                    assert m_apply_mask.called is False
+                                    assert m_cubewrite.called  # real cubewrite() should be prevented
+                                    assert m_cubewrite.call_count == 1
+                                    assert m_cubewrite.call_args_list[0].args[0] == precipitation_flux_cube
+
+
+def test_process_all_cubes_filtered(air_temp_cube, mule_vars, std_args):
+    """Ensure process() exists early if all cubes are removed in filtering."""
+    with mock.patch("mule.load_umfile"):  # ignore m_load_umfile as process_mule_vars is mocked
+        with mock.patch("umpost.um2netcdf.process_mule_vars") as m_mule_vars:
+            m_mule_vars.return_value = mule_vars
+
+            with mock.patch("iris.load") as m_iris_load:
+                section, item = split_item_code(air_temp_cube.item_code)
+                air_temp_cube.attributes = {um2nc.STASH: DummyStash(section, item)}
+                m_iris_load.return_value = [air_temp_cube]
+
+                with mock.patch("iris.fileformats.netcdf.Saver") as m_saver:  # prevent I/O
+                    m_sman = mock.Mock()
+                    m_saver().__enter__.return_value = m_sman
+
+                    infile = "/tmp/fake_input_fields_file"
+                    outfile = "/tmp/fake_input_fields_file.nc"
+                    um2nc.process(infile, outfile, std_args)
+
+                    assert m_sman.update_global_attributes.called is False
```

Comment:
This logic is assumed as a proxy for `process()` exiting early if all the cubes are filtered out.

Reply:
This is fixed with `process()` now returning a list of processed cubes (or empty list if everything is filtered out).